dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+        Logger.info(this, "Adding new 'requires_checkout_option' column to 'workflow_action' table.");
+
+        try {
+            dc.setSQL(findRequiresCheckoutOptionsColumn()).loadObjectResults();
+        } catch (Throwable e) {
+
+            Logger.info(this, "Column 'workflow_action.requires_checkout_option' does not exists, creating it");
+            needToCreate = true;
+        }
+
+        if (needToCreate) {
+            try {
+                dc.executeStatement(addRequiresCheckoutOptionColumn());
+            } catch (SQLException e) {
+                throw new DotRuntimeException("The 'requires_checkout_option' column could not be created.", e);
+            }
+        }
+    } // addRequiresCheckoutOptionColumn.
+
+    private void addSchemeIdColumn(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Adding new 'scheme_id' column to 'workflow_action' table.");
+
+        try {
+            dc.setSQL(findSchemeIdColumn()).loadObjectResults();
+        } catch (Throwable e) {
+
+            Logger.info(this, "Column 'workflow_action.scheme_id' does not exists, creating it");
+            needToCreate = true;
+        }
+        if (needToCreate) {
+            try {
+                dc.executeStatement(addSchemeIdColumn());
+            } catch (SQLException e) {
+                throw new DotRuntimeException("The 'scheme_id' column could not be created.", e);
+            }
+        }
+    } // addSchemeIdColumn.
+
+    private void createWorkflowActionStepTable(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Creating intermediate 'workflow_action_step' table.");
+
+        try {
+
+            dc.setSQL(findIntermediateTable()).loadObjectResults();
+        } catch (Throwable e) {
+
+            Logger.info(this, "Table 'workflow_action_step' does not exists, creating it");
+            needToCreate = true;
+        }
+
+        if (needToCreate) {
+            try {
+                dc.executeStatement(createIntermediateTable());
+                // The SQL Server and Oracle table definition already include de PK creation
+                if (DbConnectionFactory.isMySql() || DbConnectionFactory.isPostgres()) {
+                    dc.executeStatement(createIntermediateTablePk());
+                }
+
+                // adding the FK
+                Logger.info(this, "Creating the Workflow action step intermediate FKs.");
+                dc.executeStatement(this.createIntermediateTableForeignKeyActionId());
+                dc.executeStatement(this.createIntermediateTableForeignKeyStepId());
+            } catch (SQLException e) {
+                throw new DotRuntimeException(
+                        "The 'workflow_action_step' table could not be created.", e);
+            }
+        }
+    } // createWorkflowActionStepTable.
+
+    private boolean isLocked(final Object requiresCheckout) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "isLocked" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)